greenplumpython/func.py

```diff
@@ -146,10 +155,42 @@ class ArrayFunctionExpr(FunctionExpr):
 
     def _serialize(self) -> str:
         args_string = (
-            ",".join([f"array_agg({str(arg)})" for arg in self._args]) if any(self._args) else ""
+            ",".join(
+                [
+                    (
+                        (
+                            f"array_agg({str(self._args[i])})"  # type: ignore
+                            if (
+                                (self._group_by is None)
+                                or (
+                                    self._group_by is not None
+                                    and (self._args[i].name not in self._group_by.get_targets())
+                                )
+                            )
+                            else str(self._args[i])  # type: ignore
+                        )
+                        if isinstance(self._args[i], Expr)
+                        else to_pg_const(self._args[i])  # type: ignore
+                    )
+                    for i in range(len(self._args))
+                ]
+            )
+            if any(self._args)
+            else ""
```

Comment:
for this string does it better to split it to some lines of code.

This one-line python code is kind of difficult to read.

Reply:
Got it, split it